pkg/model/domodel/api_loadbalancer.go

```diff
@@ -67,6 +67,16 @@ func (b *APILoadBalancerModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		DropletTag: fi.String(clusterMasterTag),
 		Lifecycle:  b.Lifecycle,
 	}
+
+	if b.Cluster.Spec.NetworkID != "" {
+		loadbalancer.VPCUUID = fi.String(b.Cluster.Spec.NetworkID)
+	} else if b.Cluster.Spec.NetworkCIDR != "" {
```

Comment:
Sounds like a good idea. Updated logic to return error if both are defined. I did it in another spot where we validate the cluster spec. If code hits here, it would contain either NetworkID or NetworkCIDR.